Fix job matching without explicit Identifier keyword.

ArneBab opened the pull request:
if identifier is not given explicitly in the options, we need to add it to ensure that the replies find matching jobs.

This makes node.refstats() work again.

Commit:
disable fcpPluginMessage test, because it requires the hello world plugin loaded

Commit:
FIX: refstats stalled forever because it did not receive its reply.

The id needs to be repeated in the identifier.

Commit:
always add an Identifier field to FCP commands

Thynix commented:
In my testing (through the test-autograde script) it doesn't appear broken. Are you sure this is broken on master?

ArneBab commented:
It was broken for me, but I did quite some internal meddling in my branches which might have broken stuff…

Commit:
m3u fix: audio/mpegurl has to be audio/x-mpegurl

Commit:
do not lowercase filenames when making them urlsafe

Commit:
comment polish

Commit:
fix caching in testDDA

Commit:
actually use TestDDA again

Commit:
FIX pep8 style violations

Commit:
fix erroneously reported error with working DDA

Commit:
update version to 0.3.4 and update CHANGELOG

bertm committed:
Don't close the Node connection on JobTicket.getResult() exception

There's no need to close the connection to the node here, and this
behaviour is undocumented. The previous behaviour did not allow users
to check whether the request, when it is completed, was successful or
not: FCPGetFailed is an Exception and would cause the connection to be
shutdown.

bertm committed:
Fix documentation typos

Commit:
merge documentation fixes

ArneBab commented:
closing: changes already includen
